prefect_aws/ecs.py

```diff
@@ -371,6 +420,25 @@ def image_is_required(cls, values: dict) -> dict:
             )
         return values
 
+    @validator("task_customizations", pre=True)
+    def cast_customizations_to_a_json_patch(
+        cls, value: Union[List[Dict], JsonPatch]
+    ) -> JsonPatch:
+        if isinstance(value, list):
+            return JsonPatch(value)
+        return value
```

Comment:
Should this be in block_initialization?

I've wondered pydantic validators should change / update the value.

Reply:
This kind of validator _extends_ the supported input types. For example, Pydantic will cast items to the correct type when feasible, but it does not know how to convert a list to a `JsonPatch` type. So here, before Pydantic checks that the type is correct, we add explicit casting of lists. We can't implement this post-Pydantic validation without adjusting our type annotation.

Reply:
We do this all over the place, I think it's fair/intended use of a validator.

Reply:
That makes sense! Thanks for the explanation